src/components/inputs/z-searchbar/index.tsx

```diff
@@ -0,0 +1,429 @@
+        value={this.value}
+      />
+    );
+  }
+
+  private renderButton(): HTMLZButtonElement | null {
+    if (this.preventSubmit) {
+      return null;
+    }
+
+    return (
+      <z-button
+        variant={ButtonVariant.PRIMARY}
+        onClick={() => this.handleSubmit()}
+      >
+        CERCA
+      </z-button>
+    );
+  }
+
+  private renderResults(): HTMLDivElement | null {
+    if (
+      !this.showResults ||
+      !this.autocomplete ||
+      !this.searchString ||
+      this.searchString.length < this.autocompleteMinChars ||
+      !this.resultsItemsList ||
+      !this.resultsItemsList.length
+    ) {
+      return null;
+    }
+
+    return (
+      <div class="results">
+        <z-list
+          role="listbox"
+          id={`list-${this.htmlid}`}
+        >
+          {this.renderSearchHelper()}
+          {this.renderResultsItems()}
+          {this.renderShowAllResults()}
+        </z-list>
+      </div>
+    );
+  }
+
+  private renderResultsItems(): HTMLZListGroupElement[] {
+    const groupedItems = this.getGroupedItems(this.resultsItemsList);
+    const listGroups: HTMLZListGroupElement[] = [];
+    let counter = 0;
+
+    Object.values(groupedItems).forEach((groupItem: SearchbarGroup, index: number, array) => {
+      if (this.checkResultsCount(counter)) {
+        const listGroupsElements: HTMLZListElement[] = [];
+        groupItem.items.forEach((item: SearchbarItem, subindex: number, subarray) => {
+          if (this.checkResultsCount(counter)) {
+            const isLast = index === array.length - 1 && subindex === subarray.length - 1;
+            listGroupsElements.push(this.renderResultsItem(item, subindex, !isLast));
+          }
+          counter++;
+        });
+
+        if (listGroupsElements.length) {
+          listGroups.push(
+            <z-list-group divider-type={ListDividerType.ELEMENT}>
+              {this.renderResultsItemCategory(groupItem)}
+              {listGroupsElements}
+            </z-list-group>
+          );
+        }
+      }
+    });
+
+    return listGroups;
+  }
+
+  private renderResultsItem(item: SearchbarItem, key: number, divider: boolean): HTMLZListElementElement {
+    return (
+      <z-list-element
+        id={`list-item-${this.htmlid}-${key}`}
+        role="option"
+        tabindex={0}
+        dividerType={divider ? ListDividerType.ELEMENT : undefined}
+        clickable
+        onClickItem={() => this.handleItemClick(item)}
+      >
+        <span class={{"item": true, "ellipsis": this.resultsEllipsis, "has-category": !!item.category}}>
+          {item?.icon && (
+            <z-icon
+              class="item-icon"
+              name={item.icon}
+            />
+          )}
+          <span
+            class="item-label"
+            title={item.label}
+            innerHTML={this.renderItemLabel(item.label)}
+          />
+        </span>
+      </z-list-element>
+    );
+  }
+
+  private renderItemLabel(label: string): string {
```

Comment:
Si potrebbe forse usare questo stesso naming, togliendo quindi "results", anche per `renderResultsItem`, `renderResultsItems` e `renderResultsItemCategory`? Tanto mi pare non ci siano altri item in giro.